packages/firestore/src/local/shared_client_state.ts

```diff
@@ -724,6 +755,29 @@ export class WebStorageSharedClientState implements SharedClientState {
     this.started = false;
   }
 
+  /**
+   * Returns the "highest" online state of all active clients ("Online" takes
+   * precedence over "Offline", which takes precedence over "Unknown").
```

Comment:
So this method and comment imply that different clients may have arbitrary different OnlineStates and we use a heuristic to decide what "wins"...  But your PR comment (and the line comment buried below) explain that only the primary client will ever be Online and so really, it always wins.

So I'm not a fan of this indirect way of propagating / determining OnlineState (everybody participates but only the primary matters).  But if you keep it, please communicate what's really going on in the comments. Because as-is this is misleading.

BUUUUT...  why are we storing OnlineState per-client if the only one that matters is the primary?  Can we just have a single "OnlineState" entry in LocalStorage, have setOnlineState() update it directly, and have SyncEngine only call setOnlineState() from the primary tab?